### PR TITLE
Update MecSimCalc library to v0.2.0

### PR DESCRIPTION
**Pull request by PointlessUser**

### CODE REVIEW

File: docs/mecsimcalc-library/file-utils.md [hunk 1]
```diff
@@ -714,18 +720,23 @@ def main(inputs):
...
 ```python
 animate_plot(
-    x: np.ndarray,
-    y: np.ndarray,
-    duration: int = 5,
-    fps: int = None,
-    title: str = "y = f(x)",
-    show_axes: bool = True,
-    save_dir: str = "/tmp/temp_animation.gif",
-) -> str:
```
Comment (w3ichen) [inline docs/mecsimcalc-library/file-utils.md:728]: so you got rid of all the type hints?

File: docs/mecsimcalc-library/file-utils.md [hunk 2]
```diff
@@ -734,15 +745,19 @@ Creates an animated plot from given x and y data and returns it as an HTML image
...
+| **`y_label`**         | **str** (optional)   | The label for the y-axis. Defaults to `"y"`.                                                                                |
+| **`title`**           | **str** (optional)   | Title of the plot. Defaults to `"y = f(x)"`.                                                                                |
+| **`show_axes`**       | **bool** (optional)  | Whether to show the x and y axes. Defaults to `True`.                                                                       |
+| **`follow_tip`**      | **bool** (optional)  | Whether to follow the tip of the line as it moves along the x-axis. Defaults to `False`.                                    |
+| **`hold_last_frame`** | **float** (optional) | The duration to hold the last frame in seconds. Defaults to `1.0`.                                                          |
+| **`save_dir`**        | **str** (optional)   | The directory to temporarily. Defaults to `"/tmp/temp_animation.gif" You can only write to the tmp directory in mecsimcalc. |
```
Comment (w3ichen) [inline docs/mecsimcalc-library/file-utils.md]: Missing a backtick on `"/tmp/temp_animation.gif"

Comment (w3ichen) [reply docs/mecsimcalc-library/file-utils.md]: Incomplete sentence: "The directory to temporarily save files"?

File: docs/mecsimcalc-library/file-utils.md [hunk 3]
```diff
@@ -757,20 +772,80 @@ import numpy as np
...
+    num_points = 250,
+    initial_value = 1,
+    step_size = 0.1,
+    slider_range = (-10, 10),
+):
+```
+
+#### Description:
+
+Creates an interactive plot with a slider using Plotly, allowing the user to dynamically update the plot based on a parameter.
```
Comment (w3ichen) [inline docs/mecsimcalc-library/file-utils.md:805]: This depends on plotly, is plotly already installed with mecsimcalc or does user have to install plotly. If user needs to, then should properly have an alert in the docs

File: docs/mecsimcalc-library/images/blank_canvas.png
Comment (w3ichen) [inline docs/mecsimcalc-library/images/blank_canvas.png:1]: Could you run all the images through image compression, like https://tinypng.com to reduce file size and make it load faster?

File: docs/output/plots.md [hunk 4]
```diff
@@ -88,3 +88,28 @@ In `.to_html()`, `include_plotlyjs` must be set to `True` for the plot to be dis
 ```html
 {{ outputs.plot }}
 ```
+
+### Plotly Slider
+
+To create a plotly figure with a slider, you can use the `plot_slider` function from the `mecsimcalc` library. The `plot_slider` function takes in a function and returns the plotly figure with a slider. The function has two arguments: the first argument is the value of the slider, and the second argument is the x value.
```
Comment (w3ichen) [inline docs/output/plots.md]: Is it explained anywhere what plotly is? Would the user know hat you're referring to?

File: docs/mecsimcalc-library/plotdraw.md [hunk 5]
```diff
@@ -3,167 +3,29 @@ sidebar_label: "Plot Draw"
...
   </select>
 </div>
 
-# Plot Draw
-
 This library is designed to provide a set of functions for drawing various types of plots, arrows, segments, and shapes using Matplotlib. These functions allow for customized plotting and annotation of graphical elements.
 
 ## General
 
-### draw_arrow
```
Comment (w3ichen) [inline docs/mecsimcalc-library/plotdraw.md:20]: What happened to all these functions?

File: docs/mecsimcalc-library/plotdraw.md [hunk 6]
```diff
@@ -173,10 +35,11 @@ Creates a blank image with specified width and height, displaying a grid.
 
 #### Arguments:
 
-| Argument     | Type    | Description                                          |
-| ------------ | ------- | ---------------------------------------------------- |
-| **`width`**  | **int** | The width of the image in pixels. (Default is 1000)  |
-| **`height`** | **int** | The height of the image in pixels. (Default is 1000) |
+| Argument     | Type               | Description                                             |
+| ------------ | ------------------ | ------------------------------------------------------- |
+| **`width`**  | **int** (optional) | The width of the image in pixels. (Default is 800)      |
```
Comment (w3ichen) [inline docs/mecsimcalc-library/plotdraw.md:40]: The format you're using for default is different from the other page. In the other one you had 

> Default is `800`

Comment (w3ichen) [reply docs/mecsimcalc-library/plotdraw.md:40]: I think the brackets look nicer but having the value in backticks is good as well

File: docs/mecsimcalc-library/plotdraw.md [hunk 7]
```diff
@@ -345,69 +157,46 @@ import mecsimcalc as msc
 import mecsimcalc.plot_draw as plot_draw
 
 def main(inputs):
-    start = (100, 200)
-    end = (400, 500)
-    end_point = plot_draw.plot_segment_pixels(start, end, text="Segment", min_spacing=50)
+    plot_draw.draw_arrow((0, 0), (1, 1), thickness=2, color='red', text='Arrow', text_offset=0.1, head_width=0.1, head_length=0.1, fontsize=10)
```
Comment (w3ichen) [inline docs/mecsimcalc-library/plotdraw.md]: Maybe spread this on multiple lines

File: docs/mecsimcalc-library/plotdraw.md [hunk 8]
```diff
@@ -882,70 +565,74 @@ import mecsimcalc as msc
 import mecsimcalc.plot_draw as plot_draw
 
 def main(inputs):
-    start_point = plot_draw.plot_annotate_arrow_end((1, 1), 45, 1, text="End", text_distance=0.5, fontsize=12, text_loc={'ha': 'center', 'va': 'top'})
+    ax = plot_draw.draw_two_inclined_axes(arrow_length=1, arrow_thickness=2, text_offset=0.1, negative_y=True, negative_x=True)
```
Comment (w3ichen) [inline docs/mecsimcalc-library/plotdraw.md]: Maybe multiple lines?

**Review by w3ichen — APPROVED** [review]
Looks good. Left some minor comments, but otherwise good to merge

File: docs/mecsimcalc-library/file-utils.md (hunk 1 above)
Comment (PointlessUser) [reply docs/mecsimcalc-library/file-utils.md:728]: Yes, type hints are still in the docstring (when they hover over the function) and in the description of each function. Just thought it was too much clutter for the docs and can cause more confusion than help

File: docs/mecsimcalc-library/file-utils.md (hunk 3 above)
Comment (PointlessUser) [reply docs/mecsimcalc-library/file-utils.md:805]: it's a dependency for the library, so if you have the mecsimcalc library, it'll download plotly automatically

File: docs/mecsimcalc-library/plotdraw.md (hunk 5 above)
Comment (PointlessUser) [reply docs/mecsimcalc-library/plotdraw.md:20]: In the last meeting, we agreed there were too many similar/redundant functions, so we deleted a lot of them

File: docs/mecsimcalc-library/plotdraw.md (hunk 6 above)
Comment (PointlessUser) [reply docs/mecsimcalc-library/plotdraw.md:40]: Thanks for pointing it out, but I'll do it another time

**Pull request merged**